doc/query.md

```diff
@@ -0,0 +1,545 @@
+ageClause.put("age", gt30);
+Map<String, Object> eqMike = new HashMap<String, Object>();
+eqMike.put("$eq", "mike");
+nameClause.put("name", eqMike);
+andClause.put("$and", Arrays.<Object>asList(ageClause, nameClause));
+query.put("$or", Arrays.<Object>asList(petClause, andClause));
+```
+
+### Executing queries
+
+To find documents matching a query, use the `IndexManager` object's `find(Map<String, Object> query)` method. This returns an object that can be used in `for ( : )` loops to enumerate over the results.
+
+```java
+QueryResult result = im.find(query);
+for (DocumentRevision rev : result) {
+    // The returned revision object contains all fields for
+    // the object. You cannot project certain fields in the
+    // current implementation.
+}
+```
+
+There is an extended version of the `find` method which supports:
+
+- Sorting results.
+- Projecting fields from documents rather than returning whole documents.
+- Skipping results.
+- Limiting the number of results returned.
+
+For any of these, use 
+
+```java
+find(Map<String, Object> query,
+     long skip,
+     long limit,
+     List<String> fields,
+     List<Map<String, String>> sortDocument)
+```
+
+#### Sorting
+
+Provide a sort document to the extended version of the `find` method to sort the results of a query. 
+
+The sort document is a list of fields to sort by. Each field is represented by a map specifying the name of the field to sort by and the direction to sort.
+
+The sort document must use fields from a single index.
+
+As yet, you can't leave out the sort direction. The sort direction can be `asc` (ascending) or `desc` (descending).
+
+```java
+// sort document: [ { "name": "asc" },
+//                  { "age": "desc" } ]
+List<Map<String, String>> sortDocument = new ArrayList<Map<String, String>>();
+Map<String, String> sortByName = new HashMap<String, String>();
+Map<String, String> sortByAge = new HashMap<String, String>();
+sortByName.put("name", "asc");
+sortByAge.put("age", "desc");
+sortDocument.add(sortName);
+sortDocument.add(sortAge);
+QueryResult queryResult = im.find(query, 0, 0, null, sortDocument);
+```
+
+Pass `null` as the `sort` argument to disable sorting.
+
+#### Projecting fields
+
+Projecting fields is useful when you have a large document and only need to use a
+subset of the fields for a given view.
+
+To project certain fields from the documents included in the results, pass a list of field names to the `fields` argument. These field names:
+
+- Must be top level fields in the document.
+- Cannot use dotted notation to access sub-documents.
+
+For example, in the following document the `name`, `age` and `pet` fields could be projected, but the `species` field inside `pet` cannot:
+
+```json
+{
+    "name": "mike",
+    "age": 12,
+    "pet": { "species": "cat" }
+}
+```
+
+To project the `name` and `age` fields of the above document:
+
+```java
+List<String> fields = new ArrayList<String>();
+fields.add("name");
+fields.add("age");
```

Comment:
Done in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3